mathesar/database/base.py

```diff
@@ -2,12 +2,19 @@
 from sqlalchemy import MetaData, create_engine, inspect
 
 from mathesar.settings import mathesar_settings
+from mathesar.database.types import types
 
 APP_PREFIX = mathesar_settings["APP_PREFIX"]
 ID = f"{APP_PREFIX}id"
 
 
-engine = create_engine(
+def create_engine_with_custom_types(*args, **kwargs):
+    engine = create_engine(*args, **kwargs)
+    engine.dialect.ischema_names.update(types.CUSTOM_TYPE_DICT)
```

Comment:
Can you add a comment to this line describing what it's doing?

Reply:
Sure, I can do that.